angular/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,245 @@
+      *ngIf="isObject(value)"
+      [position]="sliderLowPosition"
+      (move)="onSliderMove('sliderLow', $event)"
+    ></cui-slider-pointer>
+    <cui-slider-pointer
+      [position]="sliderHighPosition"
+      (move)="onSliderMove('sliderHigh', $event)"
+    ></cui-slider-pointer>
+    <div #ticksContainer>
+      <span
+        *ngFor="let tickValue of scale"
+        class="cui-slider__hashlabel"
+      >{{ translateFn && translateFn(tickValue) || tickValue }}</span>
+    </div>
+  `,
+  host: {
+    'class': 'cui-slider',
+    '[class.cui-slider--disabled]': 'disabled',
+  },
+  providers: [CUI_SLIDER_VALUE_ACCESSOR]
+})
+export class SliderComponent implements AfterViewInit, OnInit, ControlValueAccessor {
+  /** @prop Determines if minimum pointer can cross over maximum pointer | false */
+  @Input() canCross: boolean = false;
+  /** @prop Set the attribute disabled to Slider | false */
+  @Input() disabled: boolean = false;
+  /** @prop Set the initial maximum value */
+  @Input() max: number;
+  /** @prop Set the initial minimum value | 0 */
+  @Input() min: number = 0;
+  /** @prop Set visual step measurement | 1 */
+  @Input() step: number = 1;
+  /** @prop Set increment of x-axis labels | 0 */
+  @Input() tick: number = 0;
+  /** @prop Function to compute layout of Slider | null */
+  @Input() translateFn: Function | null;
+  /** @prop Set either maximum pointer value or a combination of high and low pointer values | 0 */
+  @Input()
+  get value(): SliderChange | number { return this._value; }
+  set value(value) {
+    if (this._value !== value) {
+      this._value = value;
+      this.sliderLow = isObject(this.value) && (this.value as SliderChange).low || this.min;
+      this.sliderHigh = isObject(this.value) && (this.value as SliderChange).high || this.value as number;
+      this.returnCurrentValues();
+      this.changeDetectorRef.markForCheck();
+    }
+  }
+  /** Event emitted when change a pointer position */
+  @Output() readonly change: EventEmitter<SliderChange | number> = new EventEmitter<SliderChange | number>();
+
+  @ViewChild('sliderBar') private sliderBar: ElementRef;
+  @ViewChild('ticksContainer') private ticksContainer: ElementRef;
+
+  private _value: SliderChange | number = 0;
+  private _sliderLow: number;
+  private _sliderHigh: number;
+
+  get sliderLow(): number { return this._sliderLow; }
+  set sliderLow(value) { this._sliderLow = value; }
+  get sliderHigh(): number { return this._sliderHigh; }
+  set sliderHigh(value) { this._sliderHigh = value; }
+  get sliderLowPosition() {
+    return ((this.sliderLow - this.min) / (this.max - this.min)) * 100;
+  }
+  get sliderHighPosition() {
+    return ((this.sliderHigh - this.min) / (this.max - this.min)) * 100;
+  }
+
+  scale: number[];
+  selectionLeft: number;
+  selectionWidth: number;
+  isObject: Function;
+
+  private onChange = (_: any) => {};
+
+  constructor(private changeDetectorRef: ChangeDetectorRef) {}
+
+  ngOnInit() {
```

Comment:
Can we use the same method definition format? Either the method used in SliderPointer or in SliderComponent?

Reply:
Updated. Still need two arrow functions for 'this' scope purpose.